src/scanner.ts

```diff
@@ -12,35 +12,48 @@ export interface ScanResult {
   ruleName: string;
   description: string;
   total: number;
-  ids: string[];
+  resources: {id: string}[];
 }
 
 export class Scanner {
-  private _rules: Rule[] = [];
-
-  async scan(ruleType: Rule['type'], target: string) {
-    if (!this._rules.length) await this.loadRulesFromFile();
-    return this._executeRules(ruleType, target);
+  async scan(context: RuleContext, target: RuleContext['target']) {
```

Comment:
I think we should define explicit types for all of these `Class['property']` usages. It was fairly easy to keep up in the beginning, but it's no longer clear what each of these actually is, or if `Class1['target']` is the same type as `Class2['target']`, and so on